docs/key-issues/multiple_architectures.md

```diff
@@ -0,0 +1,344 @@
+      all CPU architectures for the macOS platform are accomodated in a single
+      binary artefact.
+    - macOS has developed an ecosystem where end users expect that any macOS
+      binary will run on any macOS machine, regardless of CPU architecture.
+      As a result, when building macOS apps (`.dmg` downloadable installers
+      or similar formats, produced by tools such as
+      [py2app](https://py2app.readthedocs.io) or
+      [briefcase](https://beeware.org/project/projects/tools/briefcase/)),
+      the person building the project must be accommodate all possible CPU
+      architectures where the code *could* be executed.
+    - If binary wheels are only available in "thin" format, any issues with
+      merging those wheels into fat equivalents for distribution purposes are
+      deferred to the end user. This can be problematic as it may require
+      expert knowledge about the package being merged (such as optional
+      modules or header files that may not be present in both thin artefacts).
+      Universal2 artefacts captures this knowledge by requiring the project
+      maintainers to resolve any merging issues.
+
+    Arguments against:
+
+    - `universal2` wheels are never necessary for end users installing into
+      a locally installed Python environment exclusively for their own use,
+      which is the default experience most users have with Python.
+    - The tradeoff between download size and disk space usage vs. the upside
+      for say a .dmg installer is bad - for a typical PyData stack it takes
+      hundreds of MBs per Python environment more than thin wheels, and users
+      are likely to have quite a few environments on their system at once.
+      Meaning that defaulting to `universal2` would use several GBs of disk
+      space more.
+
+        - Disk space on older MacBook Air models is 128 GB, and up to half of that
+          can be taken up by the OS and system data itself. So a few GBs can be
+          significant.
+        - Internet plans in many countries are not unlimited; almost doubling the
+          download size of wheels is a serious cost, and not desirable for any
+          user - but especially unfriendly to users in countries where network
+          infrastructure is less developed.
+
+    - In addition, it takes extra space on PyPI (examples: `universal2` wheels
+      cost an extra 81.5 MB for NumPy 1.21.4 and 175.5 MB for SciPy 1.9.1), and
+      projects with large wheels often run into total size limits on PyPI.
+    - It imposes an extra maintenance burden for each project, because separate
+      CI jobs are needed to build and test `universal2` wheels. Typically
+      projects make tradeoffs there, because they cannot support every
+      platform. And `universal2` doesn't meet the bar for usage frequency /
+      user demand here - it's well below the demand for `musllinux`,
+      `ppc64le`, PyPy, and other such platforms with still patchy support
```

Comment:
I don't read that as the intent here.

Rather, the demand for universal2 is limited to the subset of M1 mac users who aren't satisfied with `pip install` of a thin wheel into a locally active environment.

The demand for ppc64le and musllinux wheels, however, is the *entire* group of users who are using ppc64le or the musl libc.

Reply:
Indeed, I meant "all ppc64le or musllinux users" vs. "builders of universal apps for macOS". There really is very little demand for `universal2` as far as I can tell, while there's clearly a ton of users on macOS M1.

Reply:
Ok - that's a very different (and much more reasonable claim); but definitely not what I read from the way that statement is phrased.

I'll also point out there's a certain amount of post hoc ergo propter hoc here - if universal2 wheels don't exist, there's not going to be very much evidence of demand for installing them in *any* circumstance, let alone in the specific case of app distribution :-)

Reply:
I'll rephase this then to make clearer that this is the comparison, vs. app builders and not M1 users.

> I'll also point out there's a certain amount of post hoc ergo propter hoc here 

I don't know about that. It's the same as musllinux & co - when we don't provide them (which we haven't so far), people come and ask. I'm pretty certain that the folks who show up as Alpine users and ask for Alpine (so musllinux) wheels has substantially outnumbered the  folks asking for universal2 wheels because they needed them for something.

Reply:
This bullet point on the "arguments against" really congeals down to a matter of necessities:
- a previous bullet point asserted that universal2 is not *necessary* for a specific range of cases
- for ppc64le or musllinux, it is necessary to have single-architecture wheels for the current single architecture, there is no philosophical way around this